sys/net/dsm/Makefile.dep

```diff
@@ -1 +0,0 @@
-FEATURES_REQUIRED += sock_dtls
```

Comment:
why is this deleted?

Reply:
Two reasons:

- its not a feature but a Module
- that Makefile.dep was never included

I just moved the dependency to 9b856c8e1429e89925ded0ed9786900546a2bc21, I can split this one, its one of those things that just showed up during the ci tests...

Reply:
Should I split it out?

Reply:
I guess yeah, it is completely unrelated. you can expect a quick ack from me :)